app/react/Users/components/UserSidePanel.tsx

```diff
@@ -0,0 +1,267 @@
+    const updatedGroups = groups
+      .filter(group => groupIds.includes(group._id as string))
+      .map(group => ({
+        _id: group._id!,
+        name: group.name,
+      }));
+    setUserToSave({ ...userToSave, groups: updatedGroups });
+  };
+
+  const isDuplicated = (nameVal: string) =>
+    !users.find(
+      existingUser =>
+        existingUser._id !== user._id &&
+        (existingUser.username?.trim().toLowerCase() === nameVal.trim().toLowerCase() ||
+          existingUser.email?.trim().toLowerCase() === nameVal.trim().toLowerCase())
+    );
+
+  const togglePermissionList = () => {
+    setPermissionsModalOpened(!permissionsModalOpened);
+  };
+
+  return (
+    <SidePanel open={opened}>
+      <div className="sidepanel-header">
+        <Translate>{`${user._id ? 'Edit' : 'Add'} User`}</Translate>
+      </div>
+      <div className="sidepanel-body">
+        <form id="userFrom" className="user-form" onSubmit={handleSubmit(() => onSave(userToSave))}>
+          <div id="email_field" className="form-group nested-selector">
+            <label>
+              <Translate>Email</Translate>
+            </label>
+            <input
+              type="text"
+              className="form-control"
+              autoComplete="off"
+              value={userToSave.email}
+              name="email"
+              onChange={handleInputChange}
+              ref={register({
+                required: true,
+                validate: isDuplicated,
+                maxLength: 256,
+                pattern: {
+                  value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
+                  message: 'Invalid email address',
+                },
+              })}
+            />
+            {errors.email && (
+              <div className="validation-error">
+                <Icon icon="exclamation-triangle" size="xs" />
+                {errors.email.type === 'required' && <Translate>Email is required</Translate>}
+                {errors.email.type === 'validate' && <Translate>Duplicated email</Translate>}
+                {errors.email.type === 'pattern' && <Translate>Invalid email</Translate>}
+              </div>
+            )}
+          </div>
+          <div id="role_field" className="form-group nested-selector">
+            <div>
+              <label>
+                <Translate>Role</Translate>
+              </label>
+              <button
+                id="role-info"
+                className="role-info"
+                type="button"
+                onClick={togglePermissionList}
+              >
+                <Icon icon="info-circle" />
+              </button>
+            </div>
+            <select
+              name="role"
+              className="form-control"
+              onChange={handleInputChange}
+              value={userToSave.role}
+            >
+              {Object.values(UserRole).map(role => (
+                <option key={role} value={role}>
+                  {role}
```

Comment:
I'm assuming that using the translate component here didn't work, right?

Reply:
As a reminder: the team suggested that you can use the t() function with a 'false' parameter, so that it doesn't return a component.

Reply:
That's true, It doesn't work, but I translated the roles outside the render method.